lib/ronin/support/crypto/cert.rb

```diff
@@ -356,6 +357,11 @@ def self.generate(version:    2,
                             else            cert.subject
                             end
 
+          if ca
+            extensions ||= {}
+            extensions['basicConstraints'] = ['CA:TRUE', true]
```

Comment:
Maybe we should use `extensions = extensions.merge(...)` or `extensions = {'basicConstrints' => ..., **extensions}` to avoid mutating the `extensions` value, which may be another Hash variable that's passed into `Cert.generate`.

Reply:
Hmm.. but both methods will raise error if `extenions` kwarg is not passed because default value for it is `nil`.

Reply:
How about:

```ruby
extensions ||= {}
extensions = extensions.merge(...)
```

```ruby
extensions = {..., **(extensions || {})}
```

```ruby
if extensions
  extensions = extensions.merge(...)
else
  extensions = {...}
end
```

Reply:
or we could go the messy route and rewrite the extension building block of code:

```ruby
          if ca || extensions
            extension_factory = OpenSSL::X509::ExtensionFactory.new

            extension_factory.subject_certificate = cert
            extension_factory.issuer_certificate  = ca_cert || cert

            if ca
              ext = extension_factory.create_extension('CA:TRUE',true,nil)
              cert.add_extension(ext)
            end

            if extensions
              extensions.each do |name,(value,critical)|
                ext = extension_factory.create_extension(name,value,critical)
                cert.add_extension(ext)
              end
            end
          end
```

Although this isn't exactly DRY, since we repeat the `extension_factory.create_extension` and `cert.add_extension(ext)` lines.